libclamav/readdb.c

```diff
@@ -581,22 +620,74 @@ cl_error_t readdb_parse_ldb_subsignature(struct cli_matcher *root, const char *v
     return status;
 }
 
+/**
+ * @brief Parse a yara string (subsignature equivalent in yara lingo).
+ *
+ * @param root      The matcher root (engine structure containing loaded signature patterns for matching)
+ * @param virname   Name of signature that this regex subsig came from.
+ * @param hexsig    The string containing the regex
+ * @param subsig_opts Content match pattern options. See ACPATT_* macros in matcher-ac.h.
+ * @param offset    The string offset where the pattern starts
+ * @param target    The clamav target type.
+ * @param lsigid    An array of 2 uint32_t numbers: lsig_id and subsig_id. May be NULL for testing.
+ * @param options   Database options.  See CL_DB_* macros in clamav.h.
+ * @return cl_error_t
+ */
+static cl_error_t readdb_parse_yara_string(struct cli_matcher *root, const char *virname, char *hexsig, uint8_t subsig_opts,
+                                           const char *offset, uint8_t target, const uint32_t *lsigid, unsigned int options)
+{
+    cl_error_t status = CL_EPARSE;
+    cl_error_t ret;
+
+    if (strchr(hexsig, '/')) {
+        /*
+         * Looks like a pcre subsignature.
+         */
+        ret = readdb_load_regex_subsignature(root, virname, hexsig, offset, lsigid, options);
```

Comment:
Both if/else clauses end with the same thing. De-dup?

Reply:
Done.